api/librats_collect_evidence.c

```diff
@@ -12,113 +14,173 @@
 #endif
 // clang-format on
 
-rats_attester_err_t librats_collect_evidence(attestation_evidence_t *evidence, const uint8_t *hash)
+rats_attester_err_t librats_collect_evidence(const claim_t *custom_claims,
+					     size_t custom_claims_size, uint8_t **evidence_buffer,
+					     size_t *evidence_buffer_size,
+					     uint8_t **endorsements_buffer,
+					     size_t *endorsements_buffer_size)
 {
-	uint32_t hash_len = 32;
 	rats_core_context_t ctx;
 	rats_conf_t conf;
+	uint8_t *custom_claims_buffer = NULL;
+	size_t custom_claims_buffer_size = 0;
+	uint32_t hash_len = SHA256_DIGEST_LENGTH;
+	uint8_t hash[SHA256_DIGEST_LENGTH];
+	attestation_evidence_t evidence;
+	uint8_t *tmp = NULL;
+	uint8_t *p;
+	size_t evidence_size;
+	rats_attester_err_t ret;
+
+	if (!evidence_buffer || !evidence_buffer_size || !endorsements_buffer ||
+	    !endorsements_buffer_size) {
+		RATS_ERR("Bad parameters\n");
+		return RATS_ATTESTER_ERR_INVALID;
+	}
+
+	/* Initialize pointer parameters */
+	*evidence_buffer = NULL;
+	*evidence_buffer_size = 0;
+	*endorsements_buffer = NULL;
+	*endorsements_buffer_size = 0;
 
 	memset(&ctx, 0, sizeof(rats_core_context_t));
 	memset(&conf, 0, sizeof(rats_conf_t));
-	memset(evidence, 0, sizeof(attestation_evidence_t));
+	memset(&evidence, 0, sizeof(attestation_evidence_t));
 
 	conf.api_version = RATS_API_VERSION_DEFAULT;
 	conf.log_level = RATS_LOG_LEVEL_DEFAULT;
 
 	if (rats_attest_init(&conf, &ctx) != RATS_ATTESTER_ERR_NONE)
 		return RATS_ATTESTER_ERR_INIT;
-	rats_attester_err_t q_err =
-		ctx.attester->opts->collect_evidence(ctx.attester, evidence, hash, hash_len);
 
+	ret = RATS_ATTESTER_ERR_INVALID;
+	if (serialize_claims_list(custom_claims, custom_claims_size, &custom_claims_buffer,
+				  &custom_claims_buffer_size) != RATS_ERR_NONE)
+		goto err;
+
+	SHA256(custom_claims_buffer, custom_claims_buffer_size, hash);
+
+	ret = ctx.attester->opts->collect_evidence(ctx.attester, &evidence, hash, hash_len);
+	if (ret != RATS_ATTESTER_ERR_NONE)
+		goto err;
+
+	/* Get evidence length */
+	ret = RATS_ATTESTER_ERR_INVALID;
+	if (serialize_evidence(&evidence, NULL, &evidence_size) != RATS_ERR_NONE)
```

Comment:
The generated evidence format with TCG_DICE_TAGGED_EVIDENCE_CBOR_TAG  should include CBOR tag with CBOR array of CBOR bstrs: [ quote, claims ] 
please refer to https://github.com/gramineproject/gramine/pull/1039/files#diff-652d51d0e685d753930d323e690a6ef852c42a2c7e1802e809534cb49bd6a515R274